tests/andromede/test_xpansion.py

```diff
@@ -159,18 +237,142 @@ def test_generation_xpansion_single_time_step_single_scenario(
     assert output == expected_output, f"Output differs from expected: {output}"
 
 
+def test_two_candidates_xpansion_single_time_step_single_scenario(
+    generator: Component,
+    candidate: Component,
+    cluster_candidate: Component,
+) -> None:
+    """
+    As before, simple generation expansion problem on one node, one timestep and one scenario
+    but this time with two candidates: one thermal cluster and one wind cluster.
+
+    Demand = 400
+    Generator : P_max : 200, Cost : 45
+    Unsupplied energy : Cost : 1_000
+
+    -> 200 of unsupplied energy
+    -> Total cost without investment = 45 * 200 + 1_000 * 200 = 209_000
+
+    Single candidate  : Invest cost : 490 / MW; Prod cost : 10
+    Cluster candidate : Invest cost : 200 / MW; Prod cost : 10; Nb of discrete thresholds: 10; Prod per threshold: 10
+
+    Optimal investment : 100 MW (Cluster) + 100 MW (Single)
+
+    -> Optimal cost =                       490 * 100 (Single) + 200 * 100 (Cluster) -> investment
+                    + 45 * 200 (Generator) + 10 * 100 (Single) +  10 * 100 (Cluster) -> operational
+                    = 69_000 + 11_000 = 80_000
+    """
+
+    database = DataBase()
+    database.add_data("D", "demand", ConstantData(400))
+
+    database.add_data("G1", "p_max", ConstantData(200))
+    database.add_data("G1", "cost", ConstantData(45))
+
+    database.add_data("CAND", "op_cost", ConstantData(10))
+    database.add_data("CAND", "invest_cost", ConstantData(490))
+
+    database.add_data("CLUSTER", "op_cost", ConstantData(10))
+    database.add_data("CLUSTER", "invest_cost", ConstantData(200))
+    database.add_data("CLUSTER", "p_max_per_unit", ConstantData(10))
+
+    demand = create_component(model=DEMAND_MODEL, id="D")
+
+    node = Node(model=NODE_BALANCE_MODEL, id="N")
+    network = Network("test")
+    network.add_node(node)
+    network.add_component(demand)
+    network.add_component(generator)
+    network.add_component(candidate)
+    network.add_component(cluster_candidate)
+    network.connect(PortRef(demand, "balance_port"), PortRef(node, "balance_port"))
+    network.connect(PortRef(generator, "balance_port"), PortRef(node, "balance_port"))
+    network.connect(PortRef(candidate, "balance_port"), PortRef(node, "balance_port"))
+    network.connect(
+        PortRef(cluster_candidate, "balance_port"), PortRef(node, "balance_port")
+    )
+    scenarios = 1
+
+    problem = build_problem(network, database, TimeBlock(1, [0]), scenarios)
+
+    status = problem.solver.Solve()
+
+    assert status == problem.solver.OPTIMAL
+    assert problem.solver.Objective().Value() == (45 * 200) + (490 * 100 + 10 * 100) + (
+        200 * 100 + 10 * 100
+    )
+
+    output = OutputValues(problem)
+    expected_output = OutputValues()
+    expected_output.component("G1").var("generation").value = 200.0
```

Comment:
Check equality with float values with `pytest.approx` (update everywhere, even in previous unchanged tests)

Reply:
I'm working on it on another branch with other modifications to the OutputValues class. Stay tuned :)